### DISCUSSION TITLE
How to make abbreviations for a subset of aliases

### DISCUSSION THREAD

**Discussion by sandorex** [olets/zsh-abbr#89] [Q&A] [answered]
So i keep a file that has common aliases between bash and zsh, i want to add abbr without duplicating entries for both

My plan was to use function like this
```bash
a() {
    if [ "$1" = "-a" ]; then
        shift

        # abbr is only for zsh, otherwise make a normal alias
        if command -v abbr &>/dev/null; then
            abbr -S "$@"
        else
            alias "$@"
        fi
    else
        alias "$@"
    fi
}
```

Any way to do this?

**Comment by olets** [comment]
Have you tried `abbr import-aliases` https://zsh-abbr.olets.dev/commands.html#import-aliases? Something like this might work for you:

aliases file

```shell
alias one="two"
```

bashrc

```shell
source path/to/aliases/file
```

zshrc

```shell
source path/to/aliases/file

# load zsh-abbr and then

abbr -q import-aliases
```

**Reply by sandorex** [reply]
I do not want them all to be imported i want some to be regular aliases
I guess i could just write a generator function then evaluate it in the shell

**Reply by olets** [reply, marked as answer]
Maybe in .zshrc

1. source file with aliases you want abbreviations for, and load zsh-abbr (order of those two doesn't matter)
1. `abbr -q import-aliases`
1. then define other aliases (inline or by sourcing a file)

That might not exactly fit with how you use your `a`. But something like it should work

**Reply by devnoname120** [reply]
@olets What if I mass-source aliases from zsh plugins and I want to “blacklist” some aliases because they are long and unhelpful?

For example: [`gwip='git add -A; git rm $(git ls-files --deleted) 2> /dev/null; git commit --no-verify --no-gpg-sign --message "--wip-- [skip ci]"'`](https://github.com/ohmyzsh/ohmyzsh/blob/a779d6563ffb2f0093b4b74c8d5ff0982fa3e930/plugins/git/git.plugin.zsh#L110) from oh-my-zsh.

Should I always run something like `abbr erase ubuntu; abbr erase nextcloud; abbr erase vim; abbr erase cdf; abbr erase lg; abbr erase man; abbr erase redo; abbr erase vi; abbr erase zshrc; abbr erase run-help;` right after I import the aliases? Or is there some kind of blacklist that abbr follows in order to ignore some aliases during the import?

**Reply by olets** [reply]
> ignore some aliases during the import

@devnoname120 no there's no blocklist support during import.

> Should I always run

Are you saying you have `abbr -q import-aliases` in your `.zshrc`? I recommend removing it. As you've discovered, the expected use for alias import is a one-time migration. The abbreviations you imported last time, minus the ones you deleted, will still be available.

If you're using this zshrc on multiple accounts where it may not have been run before, how about including your user abbreviations files in your shared dotfiles (https://zsh-abbr.olets.dev/advanced.html#storage-and-manual-editing). That will save the time cost of running the import, and will make all of your abbreviations available not just those created from aliases.

If that's not an option, you could add an "are there any abbreviations" condition

```shell
if (( ! ${#ABBR_REGULAR_USER_ABBREVIATIONS} )); then
  abbr import-aliases
  abbr erase foo
  abbr erase bar
fi
```

(Fwiw I'd also consider refactoring abbreviations imported from third party alias plugins. They could often be shorter. For example all of OMZ's git plugin's aliases start with `g`. That's an extra keystroke. // And with abbreviations in place, could consider no longer creating some of the aliases. Less risk of running something you didn't mean to.)
